### PR TITLE
HOTFIX: psp-4210, psp-4214, psp-4215, psp-4218, psp-4220, psp-4224

### PR DESCRIPTION
**Pull request by devinleighsmith**

### CODE REVIEW

File: frontend/src/components/common/form/NotesModal.tsx [hunk 1]
```diff
@@ -53,8 +54,10 @@ export const NotesModal: React.FunctionComponent<INotesModalProps> = ({
         okButtonText="Save"
         cancelButtonText="Cancel"
         handleOk={() => {
-          onSave && onSave(values);
-          setShowNotes(false);
+          if (!error) {
```
Comment (devinleighsmith) [inline frontend/src/components/common/form/NotesModal.tsx:57]: this ensures that the notes modal does not allow a note to be saved if it exceeds the validation schema length.

File: frontend/src/features/contacts/contact/create/validation.ts [hunk 2]
```diff
@@ -36,24 +36,29 @@ const baseSchema = Yup.object().shape({
...
-    countryId: Yup.string().when('streetAddress1', {
-      is: (streetAddress1: string) => !!streetAddress1,
-      then: Yup.string().required('Country is required'),
-    }),
-    // $otherCountry - you can prefix properties with $ to specify a property that is dependent on context passed in by validate()
-    provinceId: Yup.string().when(['streetAddress1', 'countryId', '$otherCountry'], {
-      is: (streetAddress1: string, countryId: string, otherCountry: string) =>
-        !!streetAddress1 && countryId !== otherCountry,
-      then: Yup.string().required('Province/State is required'),
+  mailingAddress: Yup.object().when('useOrganizationAddress', {
```
Comment (devinleighsmith) [inline frontend/src/features/contacts/contact/create/validation.ts:39]: when we are using an organization address we should not validate these address fields.

File: frontend/src/features/contacts/contact/edit/Organization/UpdateOrganizationForm.tsx [hunk 3]
```diff
@@ -61,7 +61,22 @@ export const UpdateOrganizationForm: React.FC<{ id: number }> = ({ id }) => {
   ]);
 
   const initialValues = !!formOrganization
-    ? { ...defaultCreateOrganization, ...formOrganization }
+    ? {
+        ...defaultCreateOrganization,
```
Comment (devinleighsmith) [inline frontend/src/features/contacts/contact/edit/Organization/UpdateOrganizationForm.tsx:65]: ...overwrites the subobjects, meaning that default values weren't being set in form for undefined values in the mailing/property/billing address coming from the api.

File: frontend/src/features/leases/add/AddLeaseYupSchema.ts [hunk 4]
```diff
@@ -4,6 +4,7 @@ import * as Yup from 'yup';
 import { isLeaseCategoryVisible } from './AdministrationSubForm';
 
 export const LeaseSchema = Yup.object().shape({
+  statusType: Yup.string().required('Required'),
```
Comment (devinleighsmith) [inline frontend/src/features/leases/add/AddLeaseYupSchema.ts:7]: ensures that the choose a value option can't be selected.

File: frontend/src/features/leases/detail/LeasePages/insurance/InsuranceContainer.tsx [hunk 5]
```diff
@@ -40,11 +43,20 @@ const InsuranceContainer: React.FunctionComponent = () => {
             await refresh();
             history.push(location.pathname);
           }}
-          onCancel={() => {
-            history.push(location.pathname);
+          onCancel={(dirty?: boolean) => {
+            if (dirty) {
```
Comment (devinleighsmith) [inline frontend/src/features/leases/detail/LeasePages/insurance/InsuranceContainer.tsx:47]: properly show insurance cancel modal if cancelling when form is dirty.

File: frontend/src/features/leases/detail/LeasePages/tenant/AddLeaseTenantYupSchema.ts [hunk 6]
```diff
@@ -1,5 +1,9 @@
 import * as Yup from 'yup';
 
 export const AddLeaseTenantYupSchema = Yup.object().shape({
-  note: Yup.string().max(2000),
+  tenants: Yup.array().of(
```
Comment (devinleighsmith) [inline frontend/src/features/leases/detail/LeasePages/tenant/AddLeaseTenantYupSchema.ts:4]: schema not nesting notes field properly in previous version.

File: frontend/src/features/leases/add/ReferenceSubForm.tsx [hunk 7]
```diff
@@ -45,16 +45,6 @@ const ReferenceSubForm: React.FunctionComponent<IReferenceSubFormProps> = props
           />
         </Col>
       </Row>
-      <Row>
```
Comment (devinleighsmith) [inline frontend/src/features/leases/add/ReferenceSubForm.tsx:48]: the tfa and ps numbers are copied in their original form into the lease notes, also including them here is confusing and non-searchable

**Review by asanchezr — APPROVED** [review]

**Pull request merged**